compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineDefaultCodegen.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.jvm.codegen
+
+import org.jetbrains.kotlin.codegen.inline.MethodBodyVisitor
+import org.jetbrains.kotlin.codegen.inline.SourceMapCopier
+import org.jetbrains.kotlin.ir.declarations.IrValueParameter
+import org.jetbrains.kotlin.ir.expressions.IrExpression
+import org.jetbrains.kotlin.ir.expressions.IrFunctionAccessExpression
+import org.jetbrains.kotlin.ir.expressions.IrGetValue
+import org.jetbrains.kotlin.ir.expressions.IrTypeOperatorCall
+import org.jetbrains.org.objectweb.asm.Label
+import org.jetbrains.org.objectweb.asm.Type
+
+/**
+ * A specialization of IrInlineCodegen for calls to the underlying method in a $default handler.
+ * Such calls are inlined verbatim in the JVM backend (see InlineCodegenForDefaultBody.kt).
+ * For compatibility we have to do the same thing in the JVM IR backend.
+ */
+object IrInlineDefaultCodegen : IrInlineCallGenerator {
+    override fun genValueAndPut(
+        irValueParameter: IrValueParameter,
+        argumentExpression: IrExpression,
+        parameterType: Type,
+        codegen: ExpressionCodegen,
+        blockInfo: BlockInfo
+    ) {
+        // This codegen is only used for calls to the underlying function in a $default stub.
+        // For such calls we know that we are passing along the value parameters and reusing the same indices.
+        // There is no need to generate any code.
+        assert(argumentExpression is IrGetValue || argumentExpression is IrTypeOperatorCall && argumentExpression.argument is IrGetValue)
+    }
+
+    override fun genInlineCall(callableMethod: IrCallableMethod, codegen: ExpressionCodegen, expression: IrFunctionAccessExpression) {
+        val function = expression.symbol.owner
+        val nodeAndSmap = codegen.classCodegen.generateMethodNode(function, codegen.delegatedPropertyOptimizer)
+        val childSourceMapper = SourceMapCopier(codegen.smap, nodeAndSmap.classSMAP)
+
+        val argsSize =
+            (Type.getArgumentsAndReturnSizes(callableMethod.asmMethod.descriptor) ushr 2) - if (function.isStatic) 1 else 0
```

Comment:
Looks like there is possible to copy all variable and remove first commit (it's not necessery to update test data in first commit in any case)

Reply:
The problem with that is that we would need to change the start label (see `InlineCodegenForDefaultBody`). Both are a bit of a hack, and I thought that this solution was a bit cleaner. What do you think?

In any case, I could move the test unmute to a later commit, but given that the first commit fixes the test in question I would rather leave it there.

Reply:
>The problem with that is that we would need to change the start label (see InlineCodegenForDefaultBody). Both are a bit of a hack, and I thought that this solution was a bit cleaner. What do you think?

Oh yes, I've missed it

> In any case, I could move the test unmute to a later commit, but given that the first commit fixes the test in question I would rather leave it there.

As I see in current state (after all commits) this change is unneccessery:
-// 1 LOCALVARIABLE p\$iv Ljava/lang/String; L2 L4 0
+// 1 LOCALVARIABLE p\$iv Ljava/lang/String; L2 L\d+ 0

Reply:
> As I see in current state (after all commits) this change is unneccessery:

Thanks, I missed that! I've moved the unmute to a later commit.